pkg/logs/config/integration_config.go

```diff
@@ -214,6 +214,10 @@ func validateProcessingRules(rules []LogsProcessingRule) ([]LogsProcessingRule,
 		if rule.Name == "" {
 			return nil, fmt.Errorf("LogsAgent misconfigured: all log processing rules need a name")
 		}
+		_, err := regexp.Compile(rule.Pattern)
```

Comment:
two questions:
- how does Compile behave when the pattern is empty, which is what happens most of the time
- could we avoid compiling twice (here and in MustCompile)